NetTopologySuite.IO/NetTopologySuite.IO.PostGis/PostGisReader.cs

```diff
@@ -114,11 +108,9 @@ public IGeometry Read(Stream stream)
         /// </summary>
         /// <param name="reader"></param>
         /// <returns></returns>
-        protected IGeometry Read(BinaryReader reader)
+        protected IGeometry Read(ConfigurableBinaryReader reader)
         {
-            // Dummy read, just for bytes compatibility.
-            // The byte order is determined only once.
-            reader.ReadByte();
+            reader.EncodingType = (ByteOrder)reader.ReadByte();

```

Comment:
By itself this seems fine to me, but it also seems odd that the regular ol' `WKBReader` (and, by extension, `MsSqlSpatialReader`) just reads the byte order once at the very start and then ignores the byte order of nested geometries.  Perhaps we should have those also use this?

Reply:
After doing so, we could also consider marking `BEBinaryReader` obsolete, since those would be the last two things that use it.

Reply:
I'd say leave the signature of the functions as they were before (BinaryReader).
Add a type check (Contract? or check for argument being the required type) and at the point where we read the encoding/endianess cast to the specific reader.

Reply:
Same for `WKBReader`?

Reply:
I did not notice you changed them, too, if so, please do that as for PostGisReader.
Or did I miss your point?